### PR TITLE
fix(elastic): fix failing tests due to unsorted indexes

### LINKED ISSUE #359: fix(elasticsearch): inconsistent order of extracted records
**Describe the bug**
The indices of elasticsearch are sometimes not in ascending order, also the columns. Thus leading to failing tests on our CI

**To Reproduce**
run `make test-plugins` locally for a few times, and the test will break.

**Expected behavior**
The test should pass consistently without false-negatives.

### PR DESCRIPTION
**Pull request by GrayFlash**
closes #359

### CODE REVIEW

File: plugins/extractors/elastic/elastic.go [hunk 1]
```diff
@@ -101,22 +102,32 @@ func (e *Extractor) Extract(ctx context.Context, emit plugins.Emit) (err error)
 		return
 	}
 	x := reflect.ValueOf(r["indices"]).MapRange()
+	var indexes []string
 	for x.Next() {
 		indexName := x.Key().String()
-		docProperties, err1 := e.listIndexInfo(x.Key().String())
+		indexes = append(indexes, indexName)
+	}
+	sort.Strings(indexes)
```
Comment (StewartJingga) [inline plugins/extractors/elastic/elastic.go:110]: does the indices order also cause the flaky test, or is it just columns order?

File: plugins/extractors/elastic/elastic.go [hunk 2]
```diff
@@ -6,6 +6,7 @@ import (
 	"encoding/json"
 	"fmt"
 	"reflect"
+	"sort"
```
Comment (mabdh) [inline plugins/extractors/elastic/elastic.go:9]: @GrayFlash if it is to solve the test problem. why don't we just sort the expected and the actual data in test?

File: plugins/extractors/elastic/elastic.go [hunk 3]
```diff
@@ -101,22 +102,32 @@ func (e *Extractor) Extract(ctx context.Context, emit plugins.Emit) (err error)
...
 	}
 	x := reflect.ValueOf(r["indices"]).MapRange()
+	var indexes []string
 	for x.Next() {
 		indexName := x.Key().String()
-		docProperties, err1 := e.listIndexInfo(x.Key().String())
+		indexes = append(indexes, indexName)
+	}
+	sort.Strings(indexes)
+	for _, index := range indexes {
```
Comment (StewartJingga) [inline plugins/extractors/elastic/elastic.go:111]: let's rename `index` to `indexName`, `index` can mean another thing in a for loop.

**Review by StewartJingga — CHANGES_REQUESTED** [review]

File: plugins/extractors/elastic/elastic.go (hunk 2 above)
Comment (StewartJingga) [reply plugins/extractors/elastic/elastic.go:9]: that actually works too 🤔

File: plugins/extractors/elastic/elastic.go (hunk 1 above)
Comment (GrayFlash) [reply plugins/extractors/elastic/elastic.go:110]: yes there were some instance where it was out of order

File: plugins/extractors/elastic/elastic.go (hunk 2 above)
Comment (GrayFlash) [reply plugins/extractors/elastic/elastic.go:9]: Thought we should maintain the consistency in order of records being emmited. Usually all other extractor's client follow a certain pattern in showing the result of query.

Comment (mabdh) [reply plugins/extractors/elastic/elastic.go:9]: Hmm but if the problem is in the test case (flaky test) and there is no requirements to sort the extracted records, adding this in the main code would just add some additional unnecessary computation. 🤔

Comment (GrayFlash) [reply plugins/extractors/elastic/elastic.go:9]: @mabdh was able to fix it for indexes actually in the test part, but I can seem to find a function to get the data in `Schema` from records. There doesn’t seem to be any function in metadata model interface for it. So it is still being handled by sorting the columns before emitting. Should work?? Or else we might need some changes in proto models itself.

Comment (GrayFlash) [reply plugins/extractors/elastic/elastic.go:9]: please let me know if I'm missing something here. 😅

Comment (mabdh) [reply plugins/extractors/elastic/elastic.go:9]: In that case maybe we can add `GetSchema` here? @StewartJingga 
https://github.com/odpf/meteor/blob/5f7cd1471bc68ede73b4e1206856373d84cf654e/models/metadata.go#L11

Comment (GrayFlash) [reply plugins/extractors/elastic/elastic.go:9]: hey, folks any further comment on this? shall I move on with what I suggested?

Comment (mabdh) [reply plugins/extractors/elastic/elastic.go:9]: @GrayFlash IMO, if the purpose of the test is not in the `order of columns` or `number of extracted columns should be equal to the number provided columns`. Then it does fine to reduce the dummy data to 1.

Comment (GrayFlash) [reply plugins/extractors/elastic/elastic.go]: @StewartJingga @mabdh for now I have removed one column from dummy data and tested it for the same, if we end up adding `GetSchema`, we can fix it other way.

File: plugins/extractors/elastic/elastic.go [hunk 4]
```diff
@@ -101,22 +101,30 @@ func (e *Extractor) Extract(ctx context.Context, emit plugins.Emit) (err error)
 		return
 	}
 	x := reflect.ValueOf(r["indices"]).MapRange()
+	var indexes []string
```
Comment (StewartJingga) [inline plugins/extractors/elastic/elastic.go:104]: do we need this `indexes`? we can directly process the index in the `for x.Next()` block right?

**Review by StewartJingga — CHANGES_REQUESTED** [review]

File: plugins/extractors/elastic/elastic.go (hunk 4 above)
Comment (GrayFlash) [reply plugins/extractors/elastic/elastic.go]: yes, we can do that. removed all the changes from elastic.go btw, made them just to sort the index and columns. but they are not necessary anymore.

**Review by StewartJingga — APPROVED** [review]

**Pull request merged**